tests/nativeapp/test_teardown_processor.py

```diff
@@ -759,64 +761,70 @@ def test_drop_package_variable_mistmatch_w_special_comment_auto_drop(
 )
 def test_drop_package_variable_mistmatch_w_special_comment_quoted_name_auto_drop(
     mock_get_distribution,
-    mock_execute,
+    mock_execute_sql,
+    mock_execute_na,
     auto_yes_param,
     special_comment,
     temp_dir,
     mock_cursor,
 ):
     mock_get_distribution.return_value = "internal"
 
-    side_effects, expected = mock_execute_helper(
+    side_effects_na, expected_na = mock_execute_helper(
```

Comment:
why is this test changing so much? This should be a pure refactoring, no?

Reply:
Yes, the logic didn't change. Since not all methods of `NativeAppManager` were refactored yet, the flow in this test executes SQL via both the mixin executor, and the new SQLExecutor. So instead of a single mock, we now need two of them, and expect different calls on each.

Reply:
But then we lose fidelity here, since we can't tell if the calls happened in the right order across the mocks. Look into setting up a parent mock and using https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.attach_mock to make sure that we collect all calls across the mocks. This way, the test shouldn't need to be modified.

Reply:
Nice suggestion! Combined both mocks to a single parent.